### PR TITLE
Allow empty trigger-prefixes/trigger-patterns to coexist

### PR DESCRIPTION
**Pull request by matejrisek**
## Description

Allowing both `trigger-patterns` and `trigger-prefixes` if at least one of them is empty.

## Testing plan

To test these changes locally, create a local go project, and in the go.mod include a line similar to the following:

replace github.com/hashicorp/go-tfe => /Users/YOURUSER/go/src/github.com/hashicorp/go-tfe

Then, you can test that adding both `trigger-patterns` and `trigger-prefixes` to a workspace is allowed as long one or both are empty arrays. If you populate both with values we should raise an error.

## Output from tests
Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.

```
$ TFE_ADDRESS="https://example" TFE_TOKEN="example" TF_ACC="1" go test ./... -v -tags=integration -run TestFunctionsAffectedByChange

...
```

### CODE REVIEW

**Review by omarismail — DISMISSED** [review]

**Review by annawinkler — CHANGES_REQUESTED** [review]
Thanks for opening this PR! Could you add a few steps on how one might test this change locally? Thank you!

**Comment by matejrisek** [comment]
> Thanks for opening this PR! Could you add a few steps on how one might test this change locally? Thank you!

@annawinkler 
Hey Anna, yeah no problem, I could add steps, but could you please clarify what should be covered with them?
Do we usually add these kind of steps? Or is it because of the peculiarities of how we enable feature flags that it be better to add them?

**Comment by uturunku1** [comment]
> > Thanks for opening this PR! Could you add a few steps on how one might test this change locally? Thank you!
> 
> @annawinkler Hey Anna, yeah no problem, I could add steps, but could you please clarify what should be covered with them? Do we usually add these kind of steps? Or is it because of the peculiarities of how we enable feature flags that it be better to add them?

Hey Matej! To add to what Anna mention, we usually want contributors to add these kind of steps whenever a PR is adding a new feature or fixing a bug. Our team, the CLI team owns the project but a lot of the times we have other teams bringing in changes to the project that we lack context on. So we try to remind those contributors to fill us in as much as possible on what those changes are about and to leave instructions on how to smoke test their new functionality (this may include instructions on how to smoke test with Atlas as well, if this feature was sitting behind a feature flag).

I notice that this PR is not really bringing new functionality but just making adjustments to this previous PR: https://github.com/hashicorp/go-tfe/pull/400
So I don't think is super needed to add instructions on how we can smoke test your changes for this PR. However, for futures references, the previous PR (#400), is definitely that kind of PR were we like to see instructions on how to smoke test the new feature. Thanks!

File: workspace.go [hunk 1]
```diff
@@ -1062,7 +1062,8 @@ func (o WorkspaceCreateOptions) valid() error {
 	if o.AgentPoolID == nil && (o.ExecutionMode != nil && *o.ExecutionMode == "agent") {
 		return ErrRequiredAgentPoolID
 	}
-	if o.TriggerPrefixes != nil && o.TriggerPatterns != nil {
+	if o.TriggerPrefixes != nil && len(o.TriggerPrefixes) > 0 &&
```
Comment (uturunku1) [inline workspace.go:1065]: Curious! Why do we validate the length of these attributes for the WorkspaceCreateOptions but not the WorkspaceUpdateOptions?

Comment (matejrisek) [reply workspace.go:1065]: The reason being - I was hasty and forgot to add changes to the update part 🤦🏻‍♂️ 
Good catch!

Comment (matejrisek) [reply workspace.go:1065]: a290810956ef053e4c08c864bcf0b9c65293346a

**Comment by matejrisek** [comment]
@annawinkler @uturunku1 I've used one of the recent PRs as an example of a manual test steps. Could you please check whether that's sufficient enough? 

@uturunku1 great catch on not replicating the same logic for update as for creation :)

**Review by annawinkler — APPROVED** [review]
⭐ Thank you for the updates!

**Pull request merged**